flytekit/core/gate.py

```diff
@@ -127,6 +134,9 @@ def local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], Pr
     def local_execution_mode(self):
         return ExecutionState.Mode.LOCAL_TASK_EXECUTION
 
+    def __call__(self, *args: object, **kwargs: object) -> Union[Tuple[Promise], Promise, VoidPromise, Tuple, None]:
```

Comment:
why do we have to do this, this is already in the base task type right?

Reply:
that's right. however for now, gate does not inherit base task type. so I add __call__ instead.